src/shared/debug/launchConfiguration.ts

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as _ from 'lodash'
+import * as vscode from 'vscode'
+import { AwsSamDebuggerConfiguration } from '../sam/debugger/awsSamDebugConfiguration'
+import { AWS_SAM_DEBUG_TYPE } from '../sam/debugger/awsSamDebugger'
+
+/**
+ * Reads and writes DebugConfigurations.
+ */
+export interface DebugConfigurationSource {
```

Comment:
why abstract this?

Reply:
To avoid exposing the class and give the dev more flexibility in how they choose to use this (e.g. fakes / stubs / mocks).

Reply:
but this leads to a proliferation of abstractions. In the vast majority of cases we never use the abstractions, so it's a net cost.